File: src/main/java/eu/mshade/enderframe/item/metadata/CanPlaceOnItemStackMetadata.java
package eu.mshade.enderframe.item.metadata;

import eu.mshade.enderframe.item.MaterialKey;
import eu.mshade.enderframe.metadata.MetadataKeyValue;
import eu.mshade.enderframe.metadata.itemstack.ItemStackMetadataKey;

import java.util.List;

public class CanPlaceOnItemStackMetadata extends MetadataKeyValue<List<MaterialKey>> {

    public CanPlaceOnItemStackMetadata(List<MaterialKey> metadataValue) {
        super(ItemStackMetadataKey.CAN_PLACE_ON, metadataValue);
    }
}

File: src/main/java/eu/mshade/enderframe/item/metadata/CanPlaceOnItemStackMetadata.kt
package eu.mshade.enderframe.item.metadata

import eu.mshade.enderframe.item.ItemStackMetadataKey
import eu.mshade.enderframe.item.MaterialKey
import eu.mshade.enderframe.metadata.MetadataKeyValue

class CanPlaceOnItemStackMetadata(metadataValue: MutableSet<MaterialKey>) :
    MetadataKeyValue<MutableSet<MaterialKey>>(ItemStackMetadataKey.CAN_PLACE_ON, metadataValue)
